src/server.c

```diff
@@ -17,1657 +17,609 @@
-    qd_container_t           *container;
-    qd_log_source_t          *log_source;
-    qd_log_source_t          *protocol_log_source; // Log source for the PROTOCOL module
-    void                     *start_context;
-    sys_cond_t               *cond;
-    sys_mutex_t              *lock;
-    qd_connection_list_t      conn_list;
-    int                       pause_requests;
-    int                       threads_paused;
-    int                       pause_next_sequence;
-    int                       pause_now_serving;
-    uint64_t                  next_connection_id;
-    void                     *py_displayname_obj;
-    qd_http_server_t         *http;
-    sys_mutex_t              *conn_activation_lock;
-};
 
 #define HEARTBEAT_INTERVAL 1000
 
-ALLOC_DEFINE(qd_listener_t);
-ALLOC_DEFINE(qd_connector_t);
 ALLOC_DEFINE(qd_deferred_call_t);
-ALLOC_DEFINE(qd_connection_t);
-
-const char *MECH_EXTERNAL = "EXTERNAL";
-
-//Allowed uidFormat fields.
-const char CERT_COUNTRY_CODE = 'c';
-const char CERT_STATE = 's';
-const char CERT_CITY_LOCALITY = 'l';
-const char CERT_ORGANIZATION_NAME = 'o';
-const char CERT_ORGANIZATION_UNIT = 'u';
-const char CERT_COMMON_NAME = 'n';
-const char CERT_FINGERPRINT_SHA1 = '1';
-const char CERT_FINGERPRINT_SHA256 = '2';
-const char CERT_FINGERPRINT_SHA512 = '5';
-char *COMPONENT_SEPARATOR = ";";
-
-static const int BACKLOG = 50;  /* Listening backlog */
-
-static void setup_ssl_sasl_and_open(qd_connection_t *ctx);
-static qd_failover_item_t *qd_connector_get_conn_info(qd_connector_t *ct);
-
-/**
- * This function is set as the pn_transport->tracer and is invoked when proton tries to write the log message to pn_transport->tracer
- */
-void transport_tracer(pn_transport_t *transport, const char *message)
-{
-    qd_connection_t *ctx = (qd_connection_t*) pn_transport_get_context(transport);
-    if (ctx) {
-        // The PROTOCOL module is used exclusively for logging protocol related tracing. The protocol could be AMQP, HTTP, TCP etc.
-        qd_log(ctx->server->protocol_log_source, QD_LOG_TRACE, "[%"PRIu64"]:%s", ctx->connection_id, message);
-    }
-}
-
-void connection_transport_tracer(pn_transport_t *transport, const char *message)
-{
-    qd_connection_t *ctx = (qd_connection_t*) pn_transport_get_context(transport);
-    if (ctx) {
-        // Unconditionally write the log at TRACE level to the log file.
-        qd_log_impl_v1(ctx->server->protocol_log_source, QD_LOG_TRACE,  __FILE__, __LINE__, "[%"PRIu64"]:%s", ctx->connection_id, message);
-    }
-}
 
-/**
- * Save displayNameService object instance and ImportModule address
- * Called with qd_python_lock held
- */
-qd_error_t qd_register_display_name_service(qd_dispatch_t *qd, void *displaynameservice)
-{
+// Save displayNameService object instance and ImportModule address.
+// Called with qd_python_lock held.
+//
+// XXX But it doesn't use the _lh convention?
+qd_error_t qd_register_display_name_service(qd_dispatch_t* qd, void* displaynameservice) {
```

Comment:
Previously the doc block was formatted as [doxygen comment block](https://www.doxygen.nl/manual/docblocks.html), while the new comments aren't. Is that a plan?

Reply:
We've been pretty lax about the format of function documentation.  While I like the way the doxygen format reads in the code, I've never actually generated (or used) documentation from the source.

Way back in the beginning we designed a big chunk of what is now the router as a library intended to be public.  We no longer have that goal, but it may explain why some functions have more formal documentation than others.